src/merge_operator.rs

```diff
@@ -451,4 +454,36 @@ mod test {
         }
         assert!(DB::destroy(&opts, path).is_ok());
     }
+
+    #[test]
+    fn failed_merge_test() {
+        use crate::{Options, DB};
+
+        let path = "_rust_rocksdb_failed_merge_test";
+        let mut opts = Options::default();
+        opts.create_if_missing(true);
+        opts.set_merge_operator("test operator", test_failing_merge, None);
+        {
+            let db = DB::open(&opts, path).expect("open with a merge operator");
+            db.put(b"key", b"value").expect("put_ok");
+            let res = db.merge(b"key", b"new value");
+            match res.and_then(|_e| db.get(b"key")) {
```

Comment:
Could you explain why you use `and_then` here ?

Reply:
You could write something like:
```rust
let res = db.merge(b"key", b"new value");
assert!(res.is_ok());

match db.get(b"key") {
    Ok(val) => panic!("expected merge failure to propogate, got: {:?}", val),
    Err(e) => {
        assert!(e.as_ref().contains("Could not perform merge."));
    }
}
```

Reply:
My understanding is that RocksDB is within its rights to perform the merge operation *either* at write time or at read time (and return the error then), and so I wanted to assert that the "merge+read" operation failed a whole, without asserting where Rocks actually returned the error.

In practice the code today does surface it at read time, so I could reorganize the code that way if you preferred.